scipy/stats/tests/test_distributions.py

```diff
@@ -679,6 +679,79 @@ def test_cdf(self):
                                5.06649130e-01, 8.41442111e-01], atol=1e-6)
 
 
+class TestKappa4(TestCase):
+    def test_cdf_genpareto(self):
+        # h = 1 and k != 0 is generalized Pareto
+        x = [0.0, 0.1, 0.2, 0.5]
+        h = 1.0
+        for k in [-1.9, -1.0, -0.5, -0.2, -0.1, 0.1, 0.2, 0.5, 1.0,
+                  1.9]:
+            vals = stats.kappa4.cdf(x, h, k)
+            # shape parameter is opposite what is expected
+            vals_comp = stats.genpareto.cdf(x, -k)
+            assert_allclose(vals, vals_comp)
+
+    def test_cdf_genextreme(self):
+        # h = 0 and k != 0 is generalized extreme value
+        x = [-0.5, -0.2, -0.1, 0.0, 0.1, 0.2, 0.5]
+        h = 0.0
+        for k in [-1.9, -1.0, -0.5, -0.2, -0.1, 0.1, 0.2, 0.5, 1.0,
+                  1.9]:
+            vals = stats.kappa4.cdf(x, h, k)
+            vals_comp = stats.genextreme.cdf(x, k)
+            assert_allclose(vals, vals_comp)
+
+    def test_cdf_genlogistic(self):
+        # h = -1.0 and k != 0 is generalized logistic
+
+        # Couldn't get the generalized logistic test to work, I think
+        # mostly because of how to establish the "c" parameter for the
+        # general logistic distribution.
+        pass
+
+    def test_cdf_expon(self):
+        # h = 1 and k = 0 is exponential
+        x = [0.0, 0.1, 0.2, 0.5]
+        h = 1.0
+        k = 0.0
+        vals = stats.kappa4.cdf(x, h, k)
+        vals_comp = stats.expon.cdf(x)
+        assert_allclose(vals, vals_comp)
+
+    def test_cdf_gumbel_r(self):
+        # h = 0 and k = 0 is gumbel_r
+        x = [-0.5, -0.2, -0.1, 0.0, 0.1, 0.2, 0.5]
+        h = 0.0
+        k = 0.0
+        vals = stats.kappa4.cdf(x, h, k)
+        vals_comp = stats.gumbel_r.cdf(x)
+        assert_allclose(vals, vals_comp)
+
+    def test_cdf_logistic(self):
+        # h = -1 and k = 0 is logistic
+        x = [-0.5, -0.2, -0.1, 0.0, 0.1, 0.2, 0.5]
+        h = -1.0
+        k = 0.0
+        vals = stats.kappa4.cdf(x, h, k)
+        vals_comp = stats.logistic.cdf(x)
+        assert_allclose(vals, vals_comp)
+
+    def test_cdf_uniform(self):
+        # h = 1 and k = 1 is uniform
+        x = [0.0, 0.1, 0.2, 0.5]
+        h = 1.0
+        k = 1.0
+        vals = stats.kappa4.cdf(x, h, k)
+        vals_comp = stats.uniform.cdf(x)
+        assert_allclose(vals, vals_comp)
+
+    def test_cdf_reverse_exponential(self):
+        # h = 0.0 and k = 1.0 is reverse exponential
+
+        # scipy doesn't seem to have a reverse exponential distribution.
+        pass
```

Comment:
What's so special about this "reverse exponential"? It seems to be just `cdf \propto exp(x)` for `-\infty < x < 1`.